Sources/NIOHTTPClient/ConnectionPool.swift

```diff
@@ -0,0 +1,333 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIOHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Swift Server Working Group and the SwiftNIOHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIOHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import NIO
+import NIOConcurrencyHelpers
+import NIOHTTP1
+import NIOHTTP2
+import NIOTLS
+
+class ConnectionPool {
+    init(group: EventLoopGroup, configuration: HTTPClient.Configuration) {
+        self.eventLoopGroup = group
+        self.configuration = configuration
+    }
+
+    let eventLoopGroup: EventLoopGroup
```

Comment:
NIT: We usually have all class/struct fields defined before the `init` (here and in other places)